source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -313,6 +318,31 @@ json TransformerInstance::substring_callback(const inja::Arguments &args) const
   return input.substr(start, substring_len);
 }
 
+json TransformerInstance::replace_with_random_callback(const inja::Arguments &args) {
+    // first argument: string to modify
+  const std::string &source = args.at(0)->get_ref<const std::string &>();
+    // second argument: pattern to be replaced
+  const std::string &to_replace = args.at(1)->get_ref<const std::string &>();
+
+  return 
+    absl::StrReplaceAll(source, {{to_replace, absl::StrCat(random_for_pattern(to_replace))}});
+}
+
+std::string& TransformerInstance::random_for_pattern(const std::string& pattern) {
+  auto found = pattern_replacements_.find(pattern);
+  if (found == pattern_replacements_.end()) {
+    uint64_t random[3];
+    uint64_t high = rng_.random();
+    uint64_t low = rng_.random();
+    random[0] = low;
+    random[1] = high;
+    random[2] = 0;
```

Comment:
why do we do this? can we at least include a comment to explain?

Reply:
Added a comment. The gist is shipt wanted a 128bit long random number, base64-encoded, this is an implementation of that.

Reply:
Oh, and I removed the null at the end, as Base64::encode doesn't care if the string is null-terminated.